### PR TITLE
Client server refresh

### PR DESCRIPTION
**Pull request by BichraiX**
Added endpoint /refresh from the spec : https://spec.matrix.org/v1.11/client-server-api/#post_matrixclientv3refresh

Modified authentication to comply with the spec from /refresh.

### CODE REVIEW

**Comment by guimard** [comment]
Conflict to be resolved

File: packages/matrix-client-server/src/account/password/email/requestToken.ts [hunk 1]
```diff
@@ -60,6 +60,11 @@ const RequestToken = (clientServer: MatrixClientServer): expressAppHandler => {
           // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
         } else if (nextLink && !isValidUrl(nextLink)) {
           send(res, 400, errMsg('invalidParam', 'invalid next_link'))
+        } else if (
+          typeof sendAttempt !== 'number' ||
+          sendAttempt > 1000000000
```
Comment (guimard) [inline packages/matrix-client-server/src/account/password/email/requestToken.ts:65]: Use a constant at the beginning of file

File: packages/matrix-client-server/src/account/password/msisdn/requestToken.ts [hunk 2]
```diff
@@ -68,6 +68,11 @@ const RequestToken = (clientServer: MatrixClientServer): expressAppHandler => {
           send(res, 400, errMsg('invalidParam', 'Invalid next_link'))
         } else if (!validPhoneNumberRegex.test(dst)) {
           send(res, 400, errMsg('invalidParam', 'Invalid phone number'))
+        } else if (
+          typeof sendAttempt !== 'number' ||
+          sendAttempt > 1000000000
```
Comment (guimard) [inline packages/matrix-client-server/src/account/password/msisdn/requestToken.ts:73]: Use the same constant

File: packages/matrix-client-server/src/register/msisdn/requestToken.ts [hunk 3]
```diff
@@ -163,6 +163,11 @@ const RequestToken = (clientServer: MatrixClientServer): expressAppHandler => {
           send(res, 400, errMsg('invalidParam', 'Invalid next_link'))
         } else if (!validPhoneNumberRegex.test(dst)) {
           send(res, 400, errMsg('invalidParam', 'Invalid phone number'))
+        } else if (
+          typeof sendAttempt !== 'number' ||
+          sendAttempt > 1000000000
```
Comment (guimard) [inline packages/matrix-client-server/src/register/msisdn/requestToken.ts:168]: Same here

**Comment by guimard** [comment]
maybe the `generateToken`could be renamed `generateAndSendToken`

File: packages/matrix-client-server/src/account/password/email/requestToken.ts (hunk 1 above)
Comment (BichraiX) [reply packages/matrix-client-server/src/account/password/email/requestToken.ts]: Done

**Comment by BichraiX** [comment]
> maybe the `generateToken`could be renamed `generateAndSendToken`

Yes it's clearer, done !

**Pull request merged**